src/execution_plan/optimizations/utilize_indices.c

```diff
@@ -195,6 +209,55 @@ bool _simple_predicates(const FT_FilterNode *filter) {
 	return true;
 }
 
+static bool _validateInExpression(AR_ExpNode *exp) {
+	if(!strcasecmp(exp->op.func_name, "in") == 0) return false;
+	assert(exp->op.child_count == 2);
+	AR_ExpNode *list = exp->op.children[1];
+	if(list->type != AR_EXP_OPERAND || list->operand.type != AR_EXP_CONSTANT) return false;
+	assert(list->operand.constant.type == T_ARRAY);
+	SIValue listValue = list->operand.constant;
+	if(SIArray_Length(listValue) == 0) return false;
+	uint listLen = SIArray_Length(listValue);
+	for(uint i = 0; i < listLen; i++) {
+		// Clone value since the array might be free later.
+		SIValue v = SI_CloneValue(SIArray_Get(listValue, i));
+		// Ignore everything other than number and strings.
+		if(!(SI_TYPE(v) & (SI_NUMERIC | T_STRING | T_BOOL))) return false;
+	}
+	return true;
+}
+
+static void _transformInToOrSequence(OpFilter *filter) {
+	AR_ExpNode *inOp = filter->filterTree->exp.exp;
+	AR_ExpNode *lhs = rm_malloc(sizeof(AR_ExpNode));
```

Comment:
Please use `AR_EXP_NewOpNode`, `AR_EXP_NewDistinctOpNode`, `AR_EXP_NewVariableOperandNode`
and `AR_EXP_NewConstOperandNode` to construct the new tree.
avoid `rm_malloc` and `memcpy`